rise/static/main.js

```diff
@@ -13,82 +13,104 @@ define([
-       * can be either:
-       * 'none' - no autoselect
-       * 'code' - auto-select first code cell if any
-       * 'first' - select first cell - should be
-       * considered an experimental attempt, does
-       * not seem very helpful in real life
-       */
       auto_select: 'code',
-      /* if auto_select is not 'none', this boolean
-       * says if selection focuses on the current fragment
-       * or considers the whole slide
-       */
       auto_select_fragment: true,
+
+      // aspect
+      header: undefined,
+      footer: undefined,
+      backimage: undefined,
+      overlay: undefined,
+
+      // UI
+      toolbar_icon: 'fa-bar-chart',
+      shortcuts: {
+        'slideshow' : 'alt-r',
+        'toggle-slide': 'shift-i',
+        'toggle-subslide': 'shift-u',
+        'toggle-fragment': 'shift-f',
+        // unassigned by default
+        'toggle-note': '',
+        'toggle-skip': '',
+      },
+
+      // reveal native settings passed as-is
+      theme: 'simple',
+      transition: 'linear',
+      // xxx there might be a need to tweak this one when set
+      // by the configurator, as e.g. 'false' or 'true' will result
+      // in a string and not a boolean
+      slideNumber: true,
+      width: "100%",
+      height: "100%",
+      controls: true,
+      progress: true,
+      history: true,
       scroll: false,
       center: true,
-      autolaunch: false,
+      margin: 0.1,
+      minScale: 1.0, // we need this for codemirror to work right
     };
 
-    var config_section = new configmod.ConfigSection(
-      'livereveal',
-      {base_url: utils.get_body_data("baseUrl")});
-    config_section.load();
-
-    /* dummy empty config section to load
-     * the metadata + default as a ConfigWithDefaults object
-     */
-    var _config_section = new configmod.ConfigSection(
-      '_livereveal',
-      {base_url: utils.get_body_data("baseUrl")});
-    _config_section.load();
-
-    var final_config;
-
-    var rise_meta = Jupyter.notebook.metadata.livereveal;
-
-    if(rise_meta !== undefined && Object.keys(rise_meta).length > 0){
-      final_config = $.extend(true, default_config, rise_meta);
-      final_config = new configmod.ConfigWithDefaults(_config_section, final_config);
-      console.log("RISE metadata detected. Using ONLY RISE metadata on top of the default config. Custom config disabled.")
-    } else {
-      final_config = new configmod.ConfigWithDefaults(config_section, default_config);
-      console.log("No (or empty) RISE metadata. Using ONLY custom config (if exist) on top of the default config.")
-    }
-
-    return final_config
-
+    // 1) initialize with hardwired defaults
+    $.extend(true, complete_config, hardwired_config);
+
+    // this is a ConfigSection object as per notebook/static/services/config.js 
+    let nbext_configurator = Jupyter.notebook.config;
+    // returning this will allow further chaining of then
+    return nbext_configurator.loaded.then(
```

Comment:
So you are waiting here for the notebook config to be loaded to chain it the other events ($.extend)?

Reply:
Yes that's exactly the idea.

Reply:
Roger that!